yokozach/The-Mechanical-Forest
# Created PartyMonoSystem and Scripted New Party Creation

**Changes:**
- Created `PartyMonoSystem` and set up dependencies in the `GameManager`. This `MonoSystem` has methods for creating a new party, and some boilerplate for loading a party from a save file (Coming soon(tm)). The `PartyMonoSystem` has the prefabs for our party members added in the editor as dependencies, and new instances of these party members are instantiated when creating a new party or loading a saved party.
- Created `SaveMenuController` and messages for hiding and showing the associated `CanvasGroup`.
- Created a `SaveSlot` prefab and associated class for controlling which buttons are displayed in a `SaveSlot` `GameObject` and the prefab's behavior during gameplay.
- Created a `SaveFileContainer` class, which dynamically instantiates `SaveSlots`.

**Testing instructions:**
When running the game in the editor and clicking the "New Game" button, the `SaveMenuController` will be shown. Clicking any of the `New Game` buttons in any of the save slots should cause the game state to change to `Exploration scene, meanwhile a new party is created in the `PartyMonoSystem` in the `GameManager` core architecture. The party members do not currently have any graphical elements associated with them, and can only be observed in the scene hierarchy.

**For More Info See:**
closes #88
closes #89
